### ISSUE TITLE
Edges and Loss function

### ISSUE THREAD

**Issue by vpapaioannou** [choosehappy/PytorchDigitalPathology#6]
1. In the image attached, 

![image](https://user-images.githubusercontent.com/6065702/60452192-0174b380-9bfc-11e9-8e64-115af123a703.png)

no edge is shown, although I would expect one. If I re - run the cell, sometimes an edge is shown, sometimes is not. Is this normal? I use edge_weight = 10 to help NN to separate better one region from the other. I apply the following transformations,

INPUT IMAGE TRANS
    transforms.ToPILImage()
    , transforms.RandomVerticalFlip()
    , transforms.RandomHorizontalFlip()
    # these need to be in a reproducible order, first affine transforms and then color
    , transforms.RandomCrop(size=(patch_size,patch_size),pad_if_needed=True)
    , transforms.RandomResizedCrop(size=patch_size)
    , transforms.RandomAffine( 90, translate = (0.5, 0.5), scale = ( 0.5, 2.5))
    , transforms.RandomRotation(180)
    , transforms.ColorJitter(brightness=0, contrast=0, saturation=0, hue=.5)
    , transforms.RandomGrayscale()
    , transforms.ToTensor()

LABEL IMAGE TRANS
   transforms.ToPILImage()
   , transforms.RandomVerticalFlip()
   , transforms.RandomHorizontalFlip()
    # these need to be in a reproducible order, first affine transforms and then color
   , transforms.RandomCrop( size = ( patch_size, patch_size), pad_if_needed = True)
   , transforms.RandomResizedCrop( size = patch_size, interpolation = PIL.Image.NEAREST)
   , transforms.RandomAffine( 90, translate = (0.5, 0.5), scale = ( 0.5, 2.5))
   , transforms.RandomRotation( 180)

2. In the original paper and your code, CrossEntropyLoss is used which in Pytorch is designed for multi class classification (-log of soft max). Given that the network outputs a binary mask, shouldn't Binary Cross Entropy Loss employed instead? If so, the output should be [N, 1, H, W] instead of [N, Nclass, H, W] that is now, in order to work properly. Is this correct? What's the meaning of Nclass?

Thanks, happy July :)

**Comment by choosehappy** [comment]
sounds like a display/interpolation issue?

Depending on the interpolation method used during the rotation, and the angle of the edge, it may "disappear". Additionally, when the thumbnails are small (like here), since the edge is only 1 pixel wide, it may be visually hidden by the surrounding area. perhaps try looking at the edge image on its own in a larger format (or try saving it to a png output) and/or checking to see if any of the pixels are non-zero. 

keep in mind edge_weight is only using when computing the loss, and not when making the binary mask


regarding point 2: CrossEntropyLoss seems to work equally as well as binary and yet also provides generalizability for some of our multi-class segmentation tasks.  i've not done a formal study, or know of one, but if you have evidence comparing BCELoss to CELoss, i'd be very interested in reading it
